src/xrpld/rpc/detail/RPCHelpers.cpp

```diff
@@ -384,20 +384,23 @@ ledgerFromRequest(T& ledger, JsonContext& context)
         return getLedger(ledger, ledgerHash, context);
     }
 
-    auto const index = indexValue.asString();
+    if (indexValue.isConvertibleTo(Json::stringValue))
```

Comment:
To reduce diff, you can return `{rpcINVALID_PARAMS, "ledgerIndexMalformed"}` if this condition is not true - as far as I understand the code below will stay exactly as it was.

Reply:
Changed in [f380124](https://github.com/XRPLF/rippled/pull/5237/commits/f380124966119c33193e8b91fdb857d521f70953)